homeassistant/components/slide/cover.py

```diff
@@ -0,0 +1,121 @@
+"""Support for Go Slide slides."""
+
+import logging
+
+from homeassistant.const import ATTR_ID, ATTR_FRIENDLY_NAME
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    STATE_CLOSED,
+    STATE_OPENING,
+    STATE_CLOSING,
+    DEVICE_CLASS_CURTAIN,
+    CoverDevice,
+)
+from .const import API, DOMAIN, SLIDES
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up cover(s) for Go Slide platform."""
+
+    if discovery_info is None:
+        return
+
+    entities = []
+
+    for slide in hass.data[DOMAIN][SLIDES].values():
+        _LOGGER.debug("Setting up Slide entity: %s", slide)
+        entities.append(SlideCover(hass.data[DOMAIN][API], slide))
+
+    async_add_entities(entities)
+
+
+class SlideCover(CoverDevice):
+    """Representation of a Go Slide cover."""
+
+    def __init__(self, api, slide):
+        """Initialize the cover."""
+        self._api = api
+        self._slide = slide
+        self._id = slide["id"]
+        self._unique_id = slide["mac"]
+        self._friendly_name = slide["name"]
+
+    @property
+    def unique_id(self):
+        """Return the device unique id."""
+        return self._unique_id
+
+    @property
+    def device_state_attributes(self):
+        """Return device specific state attributes."""
+        return {ATTR_ID: self._id, ATTR_FRIENDLY_NAME: self._friendly_name}
```

Comment:
Please implement the `name` property instead of setting the friendly name state attribute.